src/shared/watchdog.c

```diff
@@ -96,6 +98,16 @@ static int open_watchdog(void) {
         return update_timeout();
 }
 
+int watchdog_set_device(char *path) {
+        watchdog_dev = strdup(path);
+        return watchdog_dev ? 0 : -ENOMEM;
+}
+
+void watchdog_free_device(void) {
+        free(watchdog_dev);
```

Comment:
`watchdog_dev = mfree(watchdog_dev)`.

Or even better, just remove this function and use `watchdog_set_device(NULL)` instead.

Reply:
maybe make watchdog_free_device() an inline static wrapper in the header file, that internally just calls watchdog_set_device(NULL)